src/v/storage/disk_log_impl.cc

```diff
@@ -723,7 +753,15 @@ disk_log_impl::make_reader(log_reader_config config) {
             config.start_offset,
             _start_offset)));
     }
-    return make_unchecked_reader(config);
+
+    if (config.start_offset > config.max_offset) {
+        auto lease = std::make_unique<lock_manager::lease>(segment_set({}));
+        auto empty = model::make_record_batch_reader<log_reader>(
+          std::move(lease), config, _probe);
+        return ss::make_ready_future<model::record_batch_reader>(
+          std::move(empty));
+    }
+    return make_cached_reader(config);
```

Comment:
do we need to have cached readers that are built based on time queries (like below this comment?)